src/main/webapp/dev/js/common/questionRubric.js

```diff
@@ -355,44 +355,33 @@ function highlightRubricCol(index, questionNum, highlight) {
 }
 
 /**
- * Moves the "weights" checkbox to the weight row if it is checked, otherwise
- * moves it to the choice row
+ * Shows the weight row if the "Choices are weighted" checkbox is checked, otherwise
+ * hides the weight row.
  *
- * @param checkbox the "weights" checkbox
+ * @param checkbox the "Choices are weighted" checkbox
  */
-function moveAssignWeightsCheckbox(checkbox) {
-    const $choicesRow = checkbox.closest('thead').find('tr').eq(0);
-    const $weightsRow = checkbox.closest('thead').find('tr').eq(1);
-    const $choicesRowFirstCell = $choicesRow.find('th').first();
+function toggleAssignWeightsRow(checkbox) {
+    const $weightsRow = checkbox.closest('form').find('thead > tr').eq(1);
     const $weightsRowFirstCell = $weightsRow.find('th').first();
 
-    const $checkboxCellContent = checkbox.closest('th').children().detach();
-
-    $choicesRowFirstCell.empty();
-    $weightsRowFirstCell.empty();
-
     if (checkbox.prop('checked')) {
-        $choicesRowFirstCell.append('Choices <span class="glyphicon glyphicon-arrow-right"></span>');
-        $weightsRowFirstCell.append($checkboxCellContent);
-        $weightsRowFirstCell.find('.glyphicon-arrow-right').show();
+        checkbox.closest('form').find('tr[id^="rubricWeights"]').show();
```

Comment:
and then u can `$weightsRow.show()`. You already get the $weightsRow. There is no need to find it again. :)